vit-servicing-station-lib/src/v0/endpoints/proposals/logic.rs

```diff
@@ -15,3 +16,36 @@ pub async fn get_proposal(
     let pool = &context.read().await.db_connection_pool;
     proposals_queries::query_proposal_by_id(id, pool).await
 }
+
+pub async fn get_proposals_by_voteplan_id_and_index(
+    query_data: ProposalsByVoteplanIdAndIndex,
+    context: SharedContext,
+) -> Result<Vec<FullProposalInfo>, HandleError> {
+    let pool = context.read().await.db_connection_pool.clone();
+    let tasks: Vec<_> = query_data
+        .into_iter()
+        .map(|proposal_query| {
+            tokio::spawn(
+                proposals_queries::query_proposals_by_voteplan_id_and_indexes(
+                    proposal_query.voteplan_id,
+                    proposal_query.indexes,
+                    pool.clone(),
+                ),
+            )
+        })
+        .collect();
+    let mut results = Vec::new();
+
+    for task in tasks {
```

Comment:
did you consider `futures::future::join_all`? not that it would make much difference, though. I guess this usually would have only a few tasks, right?